libs/beacon/src/beacon_service.cpp

```diff
@@ -242,21 +258,31 @@ BeaconService::State BeaconService::OnCollectSignaturesState()
   std::size_t random_member_index = random_number_++ % missing_signatures_from.size();
   auto        it = std::next(missing_signatures_from.begin(), long(random_member_index));
 
+  FETCH_LOG_DEBUG(LOGGING_NAME, "Get Signature shares... (index: ", index, ")");
+
   qual_promise_identity_ = Identity(*it);
   sig_share_promise_ =
       rpc_client_.CallSpecificAddress(qual_promise_identity_.identifier(), RPC_BEACON,
                                       BeaconServiceProtocol::GET_SIGNATURE_SHARES, index);
 
-  // Note: this delay is effectively how long we wait for the network event to resolve
-  state_machine_->Delay(std::chrono::milliseconds(50));
+  // Timer to wait maximally for network events
+  timer_to_proceed_.Restart(std::chrono::seconds{1});
```

Comment:
Might want to evaluate this in the future

Reply:
@ejfitzgerald

Reply:
Done